metric/unaggregated/types.go

```diff
@@ -92,17 +93,20 @@ type GaugeWithPolicies struct {
 // which determines which value field is valid. We intentionally do not use value
 // pointers and nil checks to determine which type is valid in order to avoid the GC
 // overhead of marking and sweeping the metrics.
+// NB(xichen): possibly use refcounting to replace explicit ownership tracking.
 type MetricUnion struct {
 	Type          Type
 	ID            metric.ID
 	CounterVal    int64
 	BatchTimerVal []float64
 	GaugeVal      float64
+	OwnsID        bool
```

Comment:
hm, we actually don't pool these, the sole purpose of this union thing is so that we can pass a single value around (on the stack) through a chain of components without needing to write three different functions one for each specific type (counter/timer/gauge) in each component. Pooling the unions will adds unnecessary memory / GC overhead and seems unworthy for this purpose.